src/mqt/predictor/rl/PredictorEnv.py

```diff
@@ -112,6 +115,7 @@ def step(self, action: int) -> tuple[dict[str, Any], float, bool, bool, dict[Any
         if self.state.count_ops().get("unitary"):
             self.state = self.state.decompose(gates_to_decompose="unitary")
 
+        self.state._layout = self.layout  # noqa: SLF001
```

Comment:
```suggestion
        self.state.layout = self.layout
```
There should be no real need to refer to the state with a private attribute, right?
If you change this here, you have to also change all the other occurrences.

Reply:
Unfortunately, setting the `layout` attribute itself results in `AttributeError: property 'layout' of 'QuantumCircuit' object has no setter`. Do you see any other way how to set it?

Reply:
Ah. Now I see. I think there is no alternative currently. Feel free to resolve this.